system/src/system_control_internal.cpp

```diff
@@ -77,29 +78,36 @@ particle::SystemControl::SystemControl() :
 void particle::SystemControl::processRequest(ctrl_request* req, ControlRequestChannel* /* channel */) {
     switch (req->type) {
     case CTRL_REQUEST_RESET: {
-        System.reset();
-        setResult(req, SYSTEM_ERROR_NONE);
+        setResult(req, SYSTEM_ERROR_NONE, [](int result, void* data) {
```

Comment:
really nice - I realized that before the system would reset before sending the response. This way is much better.